### PR TITLE
#169645673 accommodation likes statistics

### PR DESCRIPTION
**Pull request by alexismajya**
#### What does this PR do?

- As a user, I can see the likes/unlikes of an accommodation facility

#### Description of Task to be completed?
 - Once a user logged into the barefoot nomad, he/she can see the number of accommodation likes/unlikes

#### How should this be manually tested?

- test this `get` route: `api/v1/accommodations/accommodationId/like`

#### What are the relevant pivotal tracker stories?

[Pivotal story](https://www.pivotaltracker.com/story/show/169645673)

#### Screenshots (if appropriate)

Screen Shot 2019-11-08 at 16 34 50

#### Questions:

### CODE REVIEW

File: src/services/liking.service.js [hunk 1]
```diff
@@ -13,6 +13,15 @@ const LikingService = {
...
     return sendResult(res, 200, ` The accommodation like  status updated to ${like.isLiked}`, like);
+  },
+
+  async getLikes(req, res) {
+    const { accommodationId } = req;
+    const countLikes = await db.Likings
+      .findAndCountAll({ where: { accommodationId, isLiked: true }, raw: false });
+    const countUnlikes = await db.Likings
+      .findAndCountAll({ where: { accommodationId, isLiked: false }, raw: false });
+    return sendResult(res, 200, ` The ccommodation ${accommodationId} has:  ${countLikes.count} likes, ${countUnlikes.count} unlikes `);
```
Comment (urbanishimwe) [inline src/services/liking.service.js]: it will be useful on `frontend` if you format response `data` like this:
```
{
 like: countLikes,
 unlike: countUnlikes
}
```

**Review by urbanishimwe — CHANGES_REQUESTED** [review]

**Review by urbanishimwe — APPROVED** [review]

File: src/services/liking.service.js (hunk 1 above)
Comment (alexismajya) [reply src/services/liking.service.js]: thank you @urbanishimwe, I will highly consider it

File: src/controllers/likings.controller.js [hunk 2]
```diff
@@ -10,6 +10,15 @@ const LikingsController = {
     } catch (error) {
       return sendResult(res, 400, 'Wrong accommodation ID sent');
     }
+  },
+
+  async getAccommdationLikes(req, res) {
```
Comment (lemoissonM) [inline src/controllers/likings.controller.js:15]: Great job so far! however, it would be great if you can also provide number of likes when someone is getting single accommodations.

**Review by lemoissonM — CHANGES_REQUESTED** [review]

File: src/controllers/likings.controller.js [hunk 3]
```diff
@@ -10,6 +10,15 @@ const LikingsController = {
...
+    try {
+      const { accommodationId } = req.params;
+      await LikingService.getLikes({ accommodationId }, res);
+    } catch (error) {
+      return sendResult(res, 400, 'Wrong accommodation ID sent');
```
Comment (swaibat) [inline src/controllers/likings.controller.js:21]: great job but I think to try-catch isn't required for this get method just try to test all edge cases otherwise goo job @alexismajya

**Review by swaibat — CHANGES_REQUESTED** [review]

File: src/services/liking.service.js [hunk 4]
```diff
@@ -13,6 +13,19 @@ const LikingService = {
     }
     like.update({ isLiked: !like.isLiked });
     return sendResult(res, 200, ` The accommodation like  status updated to ${like.isLiked}`, like);
+  },
+
+  async getLikes(req, res) {
```
Comment (elchroy) [inline src/services/liking.service.js]: This is an operation with the database. We do not need to refer to `req`or `res`inside this method. Just get the count of all the `likes` belonging to a particular accommodation

File: src/services/liking.service.js [hunk 5]
```diff
@@ -13,6 +13,19 @@ const LikingService = {
...
+    const { accommodationId } = req;
+    const countLikes = await db.Likings
+      .findAndCountAll({ where: { accommodationId, isLiked: true }, raw: false });
+    const countUnlikes = await db.Likings
```
Comment (elchroy) [inline src/services/liking.service.js:21]: Good job here. But do we really need to calculate the number of `unlikes`? All we need is the number of likes, an aggregate. cc @alexismajya

File: src/services/liking.service.js [hunk 6]
```diff
@@ -13,6 +13,19 @@ const LikingService = {
...
+      .findAndCountAll({ where: { accommodationId, isLiked: false }, raw: false });
+    const data = {
+      likes: countLikes.count,
+      unlikes: countUnlikes.count
+    };
+    return sendResult(res, 200, ` The ccommodation ${accommodationId} has:  ${data.likes} likes, ${data.unlikes} unlikes `, data);
```
Comment (elchroy) [inline src/services/liking.service.js]: Since we don't need the `req` or `res`, we only have to return the number of `likes`, then return them from the controller.
- Alternatively, let the number of `likes` be included when getting on accommodation

File: src/controllers/likings.controller.js (hunk 2 above)
Comment (alexismajya) [reply src/controllers/likings.controller.js:15]: Thank you Lemoisson, this is for a single accommodation

**Review by lemoissonM — APPROVED** [review]
